src/lib.rs

```diff
@@ -851,103 +854,272 @@ pub struct RegisteredClaims {
 }
 
 #[derive(Debug, Eq, PartialEq, Clone, Copy, Default)]
-/// Options for claims time validation
+/// Options for claims presence validation
+///
+/// By default, no claims (namely `iat`, `exp`, `nbf`, `iss`, `aud`)
+/// are required, and they pass validation if they are missing.
+///
+pub struct ClaimPresenceOptions {
+    /// Whether the `iat` or `Issued At` field is required
+    pub issued_at: Presence,
+    /// Whether the `nbf` or `Not Before` field is required
+    pub not_before: Presence,
+    /// Whether the `exp` or `Expiry` field is required
+    pub expiry: Presence,
+    /// Whether the `iss` or `Issuer` field is required
+    pub issuer: Presence,
+    /// Whether the `aud` or `Audience` field is required
+    pub audience: Presence,
+    /// Whether the `sub` or `Subject` field is required
+    pub subject: Presence,
+    /// Whether the `jti` or `JWT ID` field is required
+    pub id: Presence,
+}
+
+impl ClaimPresenceOptions {
+    /// Returns a ClaimPresenceOptions where every claim is required as per [RFC7523](https://tools.ietf.org/html/rfc7523#section-3)
+    pub fn strict() -> Self {
+        use Presence::*;
+        ClaimPresenceOptions {
+            issued_at: Required,
+            not_before: Required,
+            expiry: Required,
+            issuer: Required,
+            audience: Required,
+            subject: Required,
+            id: Required,
+        }
+    }
+}
+
+#[derive(Debug, Eq, PartialEq, Clone, Copy)]
+/// Options for validating temporal claims
 ///
-/// By default, no temporal claims (namely `iat`, `exp`, `nbf`)
-/// are required, and they will pass validation if they are missing.
+///
+/// Should any temporal claims be needed, set the appropriate fields.
+///
+/// To deal with clock drifts, you might want to provide an `epsilon` error margin in the form of a
+/// `chrono::Duration` to allow time comparisons to fall within the margin.
+pub struct TemporalOptions {
+    /// Allow for some leeway for clock drifts, limited to this duration during temporal validation
+    pub epsilon: Duration,
+
+    /// Specify a time to use in temporal validation instead of `Now`
+    pub now: Option<DateTime<Utc>>,
+}
+
+impl Default for TemporalOptions {
+    fn default() -> Self {
+        TemporalOptions {
+            epsilon: Duration::seconds(0),
+            now: None,
+        }
+    }
+}
+
+#[derive(Eq, PartialEq, Clone)]
+/// Options for claims validation
+///
+/// If a claim is missing, it passes validation unless the claim is marked as required within the
+/// `claim_presence_options`.
+///
+/// By default only expiry-related claims are validated (namely `exp`, `nbf`, `iat`).
```

Comment:
Should document more on the default.

   By default, no claims are required. If there are present, only expiry-related claims are validated 
   (namely `exp`, `nbf`, `iat`) with zero epsilon and maximum age duration.

Reply:
Done

Reply:
Thanks for adding them. I think the lines are now repeated?

Line 900 and 903.